### PR TITLE
fix: gateway l07

### PR DESCRIPTION
**Pull request by kelemeno**
# What ❔

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

### CODE REVIEW

File: l1-contracts/contracts/bridgehub/Bridgehub.sol [hunk 1]
```diff
@@ -123,12 +123,12 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
 
     /// @notice To set stmDeploymetTracker, only Owner. Not done in initialize, as
     /// the order of deployment is Bridgehub, Shared bridge, and then we call this
-    function setSTMDeployer(ISTMDeploymentTracker _stmDeployer) external onlyOwner {
-        stmDeployer = _stmDeployer;
-    }
+    // function setSTMDeployer(ISTMDeploymentTracker _stmDeployer) external onlyOwner {
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridgehub/Bridgehub.sol:126]: ?

File: l1-contracts/contracts/bridgehub/Bridgehub.sol [hunk 2]
```diff
@@ -123,12 +123,12 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
...
+    //     stmDeployer = _stmDeployer;
+    // }
 
-    /// @notice To set shared bridge, only Owner. Not done in initialize, as
-    /// the order of deployment is Bridgehub, Shared bridge, and then we call this
+    /// @notice To set addresses, only Owner. Not done in initialize, as
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridgehub/Bridgehub.sol:130]: let's have a clarification of sort "addresses of some of the ecosystem contracts"

File: l1-contracts/contracts/bridgehub/IBridgehub.sol [hunk 3]
```diff
@@ -182,7 +182,7 @@ interface IBridgehub is IL1AssetHandler {
 
     function stmDeployer() external view returns (ISTMDeploymentTracker);
 
-    function setSTMDeployer(ISTMDeploymentTracker _stmDeployer) external;
+    // function setSTMDeployer(ISTMDeploymentTracker _stmDeployer) external;
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridgehub/IBridgehub.sol:185]: please delete if we dont need it

File: l1-contracts/contracts/bridgehub/Bridgehub.sol [hunk 4]
```diff
@@ -168,9 +168,9 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
 
     /// @notice To set shared bridge, only Owner. Not done in initialize, as
     /// the order of deployment is Bridgehub, Shared bridge, and then we call this
-    function setSharedBridge(address _sharedBridge) external onlyOwner {
-        sharedBridge = IL1AssetRouter(_sharedBridge);
-    }
+    // function setSharedBridge(address _sharedBridge) external onlyOwner {
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridgehub/Bridgehub.sol:171]: same here, please delete

File: l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol [hunk 5]
```diff
@@ -59,6 +59,24 @@ library DynamicIncrementalMerkle {
         return bytes32(0);
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol:59]: Still, you should fix the comment for the `setup` function, which says that calling it resets the tree

File: l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol [hunk 6]
```diff
@@ -59,6 +59,24 @@ library DynamicIncrementalMerkle {
...
     }
 
+    /**
+     * @dev Resets the tree to a blank state.
+     * Calling this function on MerkleTree that was already setup and used will reset it to a blank state.
+     */
+    function reset(Bytes32PushTree storage self, bytes32 zero) internal returns (bytes32 initialRoot) {
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol:66]: also, please document the parameters for the function

File: l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol [hunk 7]
```diff
@@ -59,6 +59,24 @@ library DynamicIncrementalMerkle {
...
+        self._nextLeafIndex = 0;
+        for (uint256 i = self._zeros.length; 0 < i; --i) {
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol:68]: you can cache `self._zeros.length`

File: l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol [hunk 8]
```diff
@@ -59,6 +59,24 @@ library DynamicIncrementalMerkle {
...
+            self._zeros.pop();
+        }
+        for (uint256 i = self._sides.length; 0 < i; --i) {
+            self._zeros.pop();
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol:72]: you probably meant `self._sides.pop()`

File: l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol [hunk 9]
```diff
@@ -59,6 +59,24 @@ library DynamicIncrementalMerkle {
...
+        }
+        self._zeros.push(zero);
+        self._zeros.pop();
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol:75]: why do we do push + pop here?

File: l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol (hunk 7 above)
Comment (kelemeno) [reply l1-contracts/contracts/common/libraries/DynamicIncrementalMerkle.sol:68]: Does that need to be cached? I thought only the check and the decrement are called multiple times?

**Pull request merged**